docs/source/tools/auth-middleware/index.rst

```diff
@@ -0,0 +1,154 @@
+.. Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Authorization Middleware
+########################
+
+The authorization middleware is currently an :doc:`Early Access Feature in Labs status </support/status-definitions>`.
+
+.. toctree::
+   :hidden:
+
+   ./oauth2
+
+How to obtain an access token for a given Daml ledger that requires authorization is defined by the ledger operator.
+The authorization middleware is an API that decouples Daml applications from these details.
+The ledger operator can provide an authorization middleware that is suitable for their authentication and authorization mechanism.
+The Daml SDK includes an implementation of an authorization middleware that supports `OAuth 2.0 Authorization Code Grant <https://oauth.net/2/grant-types/authorization-code/>`_.
+
+Features
+~~~~~~~~
+
+The authorization middleware is designed to fulfill the following goals:
+
+- Agnostic of the authentication and authorization protocol required by the identity and access management (IAM) system used by the ledger operator.
+- Allow fine grained access control via Daml ledger claims.
+- Support token refresh for long running clients that should not require user interaction.
+
+Authorization Middleware API
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An implementation of the authorization middleware must provide the following API.
+
+Obtain Access Token
+*******************
+
+The application contacts this endpoint to determine if the user is authenticated and authorized to access the given claims.
+The application must forward any cookies that it itself received in the user's request.
+The response will contain an access token and optionally a refresh token if the user is authenticated and authorized.
+Otherwise, the response will be 401 Unauthorized.
+
+HTTP Request
+============
+
+- URL: ``/auth?claims=:claims``
+- Method: ``GET``
+- Headers: ``Cookie``
+
+where
+
+- ``claims`` are the requested :ref:`Daml Ledger Claims <auth-middleware-claims>`.
+
+For example::
+
+    /auth?claims=actAs:Alice+applicationId:MyApp
+
+HTTP Response
+=============
+
+.. code-block:: json
+
+    {
+	"access_token": "...",
+	"refresh_token": "..."
+    }
+
+where
+
+- ``access_token`` is the access token to use for Daml ledger commands.
+- ``refresh_token`` (optional) can be used to refresh an expired access token on the ``/refresh`` endpoint.
```

Comment:
The `/auth` endpoint only looks for the cookie and returns whatever is stored in it. I.e. it can't return a refresh token if none is present in the cookie. So, all that the additional flag could do is make the request fail if a refresh token was requested but missing. In the current case it's up to the client to check if a refresh token was returned and decide whether to fail if it's missing or not. I don't have strong preference either way. The latter approach seems a bit simpler to me.